### PR TITLE
Solution

### PR DESCRIPTION
**Pull request by Vasyl-Poremchuk**

### CODE REVIEW

**Comment by Vasyl-Poremchuk** [comment]
image

**Comment by Vasyl-Poremchuk** [comment]
image

**Comment by Vasyl-Poremchuk** [comment]
image

**Comment by Vasyl-Poremchuk** [comment]
image

**Review by ketstap162 — CHANGES_REQUESTED** [review]
Ok! But i think you need to add tests for admin panel

**Review by ketstap162 — APPROVED** [review]
Nice job!

File: taxi/views.py [hunk 1]
```diff
@@ -59,6 +89,27 @@ class CarListView(LoginRequiredMixin, generic.ListView):
     paginate_by = 5
     queryset = Car.objects.all().select_related("manufacturer")
```
Comment (Nattalli) [inline taxi/views.py:90]: remove unnecessary `.all()` here. Also check it everywhere

File: taxi/tests/test_admin.py [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+        self.client.force_login(self.admin_user)
+        self.driver = get_user_model().objects.create_user(
+            username="driver",
+            password="driver123",
+            first_name="First",
+            last_name="Last",
+            license_number="SJE12345",
+        )
+
+    def test_driver_license_number_listed(self):
```
Comment (Nattalli) [inline taxi/tests/test_admin.py:22]: what about _old but gold_ type annotations? 😺

Comment (Nattalli) [reply taxi/tests/test_admin.py:22]: add it in every function

File: taxi/tests/test_forms.py [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+        self.form_data = {
+            "license_number": "UTY89424",
+        }
+        form = DriverLicenseUpdateForm(
+            data=self.form_data,
+        )
+        self.assertTrue(form.is_valid())
+        self.assertEqual(
+            form.cleaned_data, self.form_data,
+        )
```
Comment (Nattalli) [inline taxi/tests/test_forms.py:35]: add test with invalid form data and check that license doesn't update

**Review by Nattalli — CHANGES_REQUESTED** [review]

**Review by Y-Havryliv — APPROVED** [review]
Nice!

**Pull request closed without merging**